src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+    have a encoding flag option
+    */
+
+    public function init() {
+        setSourceDirectory( 'src\main\java\' );
+        setClassOutputDirectory( 'classes\java\main\' );
+        setVerbose( false );
+        setEncode( '' );
+		setSourcePaths( [''] );
+		setRecursive( false );
+		setCreateJar( false );
+		setLibsDir( 'libs' );
+		setCompileOptionsString( '' );
+		setJarOptionsString( '' );
+		setJarNameString( 'output.jar' );
+        return this;
+    }
+
+	/**
+	 * Sets the directory to run the command in
+  	 **/
+	function projectRoot( required projectRoot ) {
+		setProjectRoot( fileSystemutil.resolvePath( projectRoot ) );
+		return this;
+	}
+
+	function fromSource( required any sourcePaths ) {
+		if( isSimpleValue( arguments.sourcePaths ) ) {
+			arguments.sourcePaths = listToArray( arguments.sourcePaths, ",", true );
+		}
+		arguments.sourcePaths = arguments.sourcePaths.map( function( s ) {
+			return fileSystemutil.resolvePath( arguments.s, getProjectRoot() );
+		} );
+		variables.sourcePaths = arguments.sourcePaths;
+		return this;
+	}
+
+    function toClasses( required classOutputDirectory ){
+        setClassOutputDirectory( fileSystemutil.resolvePath( classOutputDirectory, getProjectRoot() ) )
+        return this;
+    }
+
+    function withVerbose() {
+        setVerbose( true );
+        return this;
+    }
+
+    function withEncoding( required encodeValue ) {
+        setEncode( encodeValue );
+        return this;
+    }
+
+	function recursive() {
+		setRecursive( true );
+		return this;
+	}
+
+	function toJar( string jarName='' ) {
+		setJarNameString( jarName );
+		setCreateJar( true );
+		return this;
+	}
+
+	function libsDir( required libsDir ) {
+		setLibsDir(libsDir);
+		return this;
+	}
+
+	function compileOptions( required string options ) {
+		setCompileOptionsString( options );
+		return this;
+	}
+
+	function jarOptions( required string options) {
+		setJarOptionsString( options );
+		return this;
+	}
+
+    function run() {
+		job.start( 'Compiling' );
+
+        job.start( 'find jdk bin directory' );
+		setJavaBinFolder( findJDKBinDirectory() );
+        job.complete( getVerbose() );
+
+        job.start( 'compiling the code' );
+		compileCode();
+        job.complete( getVerbose() );
+
+		if( getCreateJar() ) {
+            job.start( 'creating the jar' );
+    		buildJar();
+            job.complete( getVerbose() );
+		}
+
+		job.complete( getVerbose() );
+    }
+
+	function compileCode() {
+
+		if( directoryExists( getClassOutputDirectory() ) ){
+			directoryDelete( getClassOutputDirectory(), true )
+		}
+
+		//shell.printString( " glob-> start... " );
+        //job.addLog( " glob-> start... " );
+
+		var globber = wirebox.getInstance( 'globber' );
+		var tempSrcFileName = tempDir & 'temp#createUUID()#.txt';
+
+		setSourcePaths( getSourcePaths().map( function( p ) {
+			var currentPath = fileSystemutil.resolvePath( arguments.p, getProjectRoot() );
+			if ( directoryExists( currentPath ) ) {
```

Comment:
Do you have the notes on how I had said to make this work?  It wasn't what you have here.

Reply:
@bdw429s dk if this is more or less what we talked: use only file globber on fromsource

check if its a folder or a file
if its a file process if not leave it alone

*.java 	non recursive

if its a folder then add **.java
**.java recursive

globber with filter out folders, to get only files